specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -485,9 +485,11 @@ export function Dialog({
         id={id('handle')}
       >
         <div className="flex items-center gap-2">
-          {typeof defaultIcon === 'object' && showIcon
-            ? defaultIcon
-            : dialogIcons[iconType]}
+          <span className="text-blue-500">
+            {typeof defaultIcon === 'object' && showIcon
+              ? defaultIcon
+              : dialogIcons[iconType]}
+          </span>
```

Comment:
after this change the code is always displaying a `<span>` even when it's empty, which causes the "gap-2" in the parent flex container to add some padding to the right of the header even when not needed

instead, define a variable like this:
```tsx
  const resolvedIcon = typeof defaultIcon === 'object' && showIcon
    ? defaultIcon
    : dialogIcons[iconType];
```

and then use it to only render span when needed:

```suggestion
          {resolvedIcon === undefined ? undefined : <span className="text-blue-500">{resolvedIcon}</span>}
```

Reply:
oh, I just realized that the only case when icon is not displayed is when you are in a Resource view dialog, or when user disabled the icon in settings - still, let's fix this bug